src/gpodder/model.py

```diff
@@ -920,7 +949,14 @@ def remove_downloaded(self):
 
     @property
     def cover_file(self):
-        return os.path.join(self.save_dir, 'folder')
+        filename = 'folder'
+        try:
+            url = urlparse(self.cover_url)
+            filename = os.path.basename(url.path)
```

Comment:
I'm kind of on the fence, I tend to agree with you but I was also thinking "why do I even care about the extension?" ie. going the other direction, store it with provided filename without all the checks on whether or not the format and the extension match and whether or not it's supported.
The reason is that at least in QML it will figure out the type of its' source by itself and the extension is irrelevant. However I can't speak to the accuracy of that statement for other gpodder platforms.
From a security perspective it may still be good to match the magicbytes to the extenstion since a file with a wrong extension may be purposefully prepared to attack a system.

> Also how do you deal with cover art with changing query parameter? basename would be identical

Do you mean example.com/images.php?id=1 and later example.com/images.php?id=101 ?
That would fail right now I think I'd be glad to hear suggestions, I was thinking that maybe an update to episode_art_url should trigger a forced download instead of the current method that skips if the file exists.